Source/WebKit/UIProcess/Cocoa/UIDelegate.mm

```diff
@@ -448,6 +449,22 @@
         return;
     }
 
+    // Some sites have quirks where multiple login domains require storage access.
+    auto additionalLoginDomain = WebCore::NetworkStorageSession::findAdditionalLoginDomain(currentDomain, requestingDomain);
+
+    if (organizationStorageAccessPromptQuirk || additionalLoginDomain) {
+        if (m_uiDelegate->m_delegateMethods.webViewRequestStorageAccessPanelWithoutQuirksUnderCurrentDomainCompletionHandler) {
```

Comment:
Technically this method is invoked when there are quirk, so we probably want to name it 
`_webView:reuqestStorageAccessPanelFromQuirksForDomains:underCurrentDomain:completionHandler:`, and we should include the quirked domains so client (TestRunnner) could verify the domains are expected.